gateway-js/src/index.ts

```diff
@@ -346,6 +359,36 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private issueDeprecationWarningsIfApplicable() {
+    // TODO(trevor:removeServiceList)
+    if ('experimental_updateSupergraphSdl' in this.config) {
+      this.logger.warn(
+        'The `experimental_updateSupergraphSdl` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('experimental_updateServiceDefinitions' in this.config) {
+      this.logger.warn(
+        'The `experimental_updateServiceDefinitions` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('serviceList' in this.config) {
+      this.logger.warn(
+        'The `serviceList` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
```

Comment:
Probably mention IntrospectAndCompose here and in the localServiceList warning?

Reply:
45c9cdaa

`localServiceList` is undocumented. I went ahead and exported `LocalCompose` (its current replacement) and mentioned in the warning message. Let me know if you feel strongly about this needing documentation.